### ISSUE TITLE
make util.ba2hex returns string instead of bytes or change the output value

### ISSUE THREAD

**Issue by cjtk** [ilanschnell/bitarray#94]
python 3.6
The problem is with the confusing way python handles the type bytes

consider the following example

from bitarray import bitarray
from bitarray import util
result = util.ba2hex(bitarray('0000'))
result
b'0'

it seems to be okay, 0000 gets convert to hex 0
the problem is in python, this 0 is not the value 0 but the ascii '0'

result[0]
48
result == b'\x30'
True

So 0000 got converted into hex 30

I feel that it is a bit misleading and I would suggest that the method returns a string instead of bytes, if it is to return bytes then '0000' should be converted to b'\x0' instead of b'\x30'

**Comment by ilanschnell** [comment]
I agree, that the current behavior is confusing.  When I implementing`ba2hex`, I looked at the standard library function `binascii.hexlify`:
```
>>> import binascii
>>> binascii.hexlify(b'\x24\x7a')
b'247a'
```
This function also returns bytes.  I assume this has historic reasons because in Python 2 bytes and strings are the same.

**Comment by ilanschnell** [comment]
I've made the change, `util.ba2hex` now always returns a string object (this used to be the case for Python 2 already).  The change is part of this commit: 8a5ebbd95da116c7b6b13522625a2b63fca6eebb
This change will be part of the next release.  Thanks!

**Comment by cjtk** [comment]
thanks heaps, this will ensure compatibility between python 2 and 3 too

**Comment by ilanschnell** [comment]
I just released bitarray 1.2.2 which contains this change.
